### PR TITLE
Lesson 6

### PR DESCRIPTION
**Pull request by OmegaLord**

### CODE REVIEW

File: lesson-3/object_finder.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module ObjectFinder
+  private
+
+  def find(array, attr_name, input)
+    array.find { |e| e.instance_variable_get("@#{attr_name.to_sym}") == input }
```
Comment (AZatsepa) [inline lesson-3/object_finder.rb]: Can we use getter instead of `instance_variable_get`?

File: mypatch.patch [hunk 2]
```diff
@@ -0,0 +1,269 @@
+diff --git a/Gemfile b/Gemfile
```
Comment (AZatsepa) [inline mypatch.patch]: ?)

File: spec/cargo_train_spec.rb [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
 describe CargoTrain do
-  let!(:cargo_train) { described_class.new('number') }
+  let(:cargo_train) { build(:cargo_train) }
   let(:cargo_carriage) { CargoCarriage.new }
-  let(:type) { :cargo }
 
   describe '#type' do
+    let(:type) { :cargo }
+
     it { expect(cargo_train.type).to eq(type) }
```
Comment (AZatsepa) [inline spec/cargo_train_spec.rb]: `it { expect(cargo_train.type).to eq(:cargo) }`

File: spec/passenger_train_spec.rb [hunk 4]
```diff
@@ -11,11 +10,15 @@
   end
 
   describe '#hook_carriage' do
+    let(:passenger_carriage) { PassengerCarriage.new }
+
     it {
       expect { passenger_train.hook_carriage(passenger_carriage) }
```
Comment (AZatsepa) [inline spec/passenger_train_spec.rb]: Shared examples?

File: spec/station_spec.rb [hunk 5]
```diff
@@ -20,4 +20,9 @@
   describe '.all' do
     it { expect(described_class.all).to include(station, station_2) }
   end
+
+  it_behaves_like 'validable' do
```
Comment (AZatsepa) [inline spec/station_spec.rb]: validatable

File: spec/support/validable.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+shared_examples_for 'validable' do
+  it { expect(valid_object).to be_valid }
+
+  it do
+    expect do
+      invalid_object
```
Comment (AZatsepa) [inline spec/support/validable.rb]: InvalidObjectClass.new(invalid name)

File: lesson-3/object_finder.rb (hunk 1 above)
Comment (OmegaLord) [reply lesson-3/object_finder.rb]: Metaprogramming...
Train -> number
Station -> name
Or can I write like this e.attr_name?

Comment (AZatsepa) [reply lesson-3/object_finder.rb]: > Metaprogramming...
> Train -> number
> Station -> name
> Or can I write like this e.attr_name?

You can write a new getter

Comment (OmegaLord) [reply lesson-3/object_finder.rb]: Something like alias for number and name?

File: lesson-3/object_finder.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module ObjectFinder
+  private
+
+  def find(array, attr_name, input)
+    raise 'This object already exists' if array.find { |e| e.public_send(attr_name.to_s) == input }
```
Comment (AZatsepa) [inline lesson-3/object_finder.rb]: .to_s is unnecessary

Comment (OmegaLord) [reply lesson-3/object_finder.rb]: I change the parameter form 'name' to name. And this is wron without .to_s.

**Pull request merged**